### DISCUSSION TITLE
How do I handle sharing/downlaods?

### DISCUSSION THREAD

**Discussion by JKDos** [android-js/androidjs#230] [Q&A] [answered]
In a webview app, if I have an image, how can I download it to the user's phone or open the share menu for them to share on social media?

I am using these permissions, but not able to download
```
 	       "android.permission.WRITE_EXTERNAL_STORAGE",
		"android.permission.WRITE_MEDIA_STORAGE",
		"android.permission.READ_EXTERNAL_STORAGE"
```

**Comment by Chhekur** [comment, marked as answer]
@JKDos
- Downloading Image
In order to download image what you can do is you can get the public URL of image and send that to `back process` over IPC and there you can the image from the received URL and can use FS module to save that in user's phone.

- Share on social media
You can create social media sharing dialog in HTML itself or React and can read from user's device using FS and share.

**Reply by JKDos** [reply]
Thanks. I actually didn't know there was sharing API in Javascript/HTML. That's great!

**Reply by JKDos** [reply]
navigator.share appears to be undefined. I assume it's unsupported for now.
